src/main/java/io/hops/util/CredentialsProvider.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.hops.util;
+
+import io.hops.util.cloud.Credentials;
+import io.hops.util.exceptions.CloudCredentialException;
+import io.hops.util.exceptions.HTTPSClientInitializationException;
+import io.hops.util.exceptions.JWTNotFoundException;
+import org.apache.spark.sql.SparkSession;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * AWS temporary credential provider.
+ */
+public class CredentialsProvider {
+  private static final Logger LOG = Logger.getLogger(CredentialsProvider.class.getName());
+  
+  private CredentialsProvider() {
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration
+   * @param role
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role) throws CloudCredentialException {
+    return assumeRole(role, 3600);
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration
+   * @param role
+   * @param durationSeconds
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role, int durationSeconds) throws CloudCredentialException {
+    String uniqueID = UUID.randomUUID().toString();
+    return assumeRole(role, "session_" + uniqueID, durationSeconds);
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration and system properties.
+   * @param role
+   * @param roleSessionName
+   * @param durationSeconds
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role, String roleSessionName, int durationSeconds)
+    throws CloudCredentialException {
+    Response response;
+    try {
+      HashMap<String, Object> queryParams = new HashMap<>();
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_ROLE, role);
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_SESSION, roleSessionName);
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_SESSION_DURATION, durationSeconds);
+      response = Hops.clientWrapper(
+          "/" + Constants.HOPSWORKS_CLOUD_RESOURCE + "/" + Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE,
+          HttpMethod.GET, queryParams);
+    } catch (HTTPSClientInitializationException | JWTNotFoundException e) {
+      throw new CloudCredentialException(e.getMessage());
+    }
+    LOG.log(Level.INFO, "******* response.getStatusInfo():" + response.getStatusInfo());
```

Comment:
Maybe Level.FINE?

Reply:
This I copied from other methods I do not even know why it is used. I have not seen this message anywhere.

Reply:
Let's remove the log messages from this PR then